### PR TITLE
breaking sqlcipher usage update

### PR DESCRIPTION
**Pull request by neonphog**
### Summary

- upgrade rusqlite
- pin cipher_compatibility to version 4
- apply ios workaround
- track salt in config, and pragma as consequence of ios workaround

### TODO:
- [x] CHANGELOG(s) updated with appropriate info

### CODE REVIEW

File: crates/lair_keystore/src/store_sqlite.rs [hunk 1]
```diff
@@ -584,16 +625,29 @@ fn encrypt_unencrypted_database(
...
+            )
+            .map_err(one_err::OneErr::new)?;
+        }
+
+        conn.execute_batch(
+            r#"
+PRAGMA encrypted.cipher_compatibility = 4;
+PRAGMA encrypted.cipher_plaintext_header_size = 32;
+"#,
         )
```
Comment (ThetaSinner) [inline crates/lair_keystore/src/store_sqlite.rs:645]:
```suggestion
configure_encryption(conn)
```

File: crates/lair_keystore/src/store_sqlite.rs [hunk 2]
```diff
@@ -584,16 +625,29 @@ fn encrypt_unencrypted_database(
...
+                "ATTACH DATABASE :db_name AS encrypted KEY :key",
+                rusqlite::named_params! {
+                    ":db_name": encrypted_path.to_str(),
+                    ":key": &lock[14..81],
+                },
+            )
+            .map_err(one_err::OneErr::new)?;
+        }
+
+        conn.execute_batch(
```
Comment (ThetaSinner) [inline crates/lair_keystore/src/store_sqlite.rs:640]: There's a test for migration that will open the newly encrypted database with the same compatibility settings right?

Just thinking ahead to when we move versions and only remember to update one :)

**Review by ThetaSinner — APPROVED** [review]
I've had a look at the sqlcipher docs and I think I understand what this is doing. Looks good to me!

File: crates/lair_keystore/src/store_sqlite.rs (hunk 2 above)
Comment (neonphog) [reply crates/lair_keystore/src/store_sqlite.rs:640]: (and yes, there's a `migrate_unencrypted` test that triggers this code to be run.)

**Pull request merged**